### DISCUSSION TITLE
Can't access Alpine.$data

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3601] [1. Help]
```
<form x-data="{ first_name:'', last_name:'' }" method="POST" @submit.prevent="submitForm">
```

I have the inputs as x-model="first_name"

```
let submitForm = () =>
{
    const formData = new FormData();
    const xData = Alpine.$data;
    console.log("xData = ", xData);
};
```
Is printing`ƒ Ee(e){return j($(e))}`

**Comment by ekwoka** [comment]
Because it's a function. the first argument is the node you want to get the Data context of.

**Reply by anjanesh** [reply]
So how do I get the values of first_name and last_name from a global function outside the component ?

**Reply by SimoTod** [reply]
Why do you need something like that? Just use Alpine. Put submitForm in your x-data

**Reply by anjanesh** [reply]
_Put submitForm in your x-data_ - I see - thanks.

**Reply by ekwoka** [reply]
You could pass in the node thats in the component.

Or use a store.

But yeah, best would probably be to just have form submission within the Alpine context.

**Comment by anjanesh** [comment]
I just found out that `Alpine.$data;` should've been `this.$data;`
```
const xData = this.$data;
console.log("first_name = ", xData.first_name);
console.log("last_name= ", xData.last_name);
```

**Reply by anjanesh** [reply]
It looks like one can access the alpine local variable if the containing element is having an id.
```
<form id="frm" x-data="{ first_name:'', last_name:'' }" method="POST" @submit.prevent="submitForm">
```

```
document.getElementById('frm')._x_dataStack[0].first_name = "john";
```

**Reply by ekwoka** [reply]
You can, it doesn't need an id.

But it would be better to do `Alpine.$data(document.getElementById('frm'))` if you must

**Reply by anjanesh** [reply]
Without an id how do I access `Alpine.$data(document.getElementById('frm'))` ?

**Reply by ekwoka** [reply]
You just get the element however you want. querySelector for example.

Or $el since it's a form.
